src/app/modules/main/wallet_section/collectibles/module.nim

```diff
@@ -81,49 +81,36 @@ method currentCollectibleModuleDidLoad*(self: Module) =
   self.checkIfModuleDidLoad()
 
 method fetchOwnedCollectibles*(self: Module) =
-  self.controller.fetchOwnedCollectibles(self.chainId, self.address)
+  self.controller.fetchOwnedCollectibles(self.chainId, self.addresses)
 
 method filterChanged*(self: Module, addresses: seq[string], chainIds: seq[int]) =
   let network = self.controller.getNetwork()
-  let account = self.controller.getWalletAccountByAddress(addresses[0])
   self.chainId = network.chainId
-  self.address = account.address
-  self.currentCollectibleModule.setCurrentAddress(network, self.address)
+  self.addresses = addresses
+  self.currentCollectibleModule.setCurrentNetwork(network)
+  self.view.setCollectibles(@[])
+  let data = self.controller.getOwnedCollectibles(self.chainId, self.addresses)
 
-  let data = self.controller.getOwnedCollectibles(self.chainId, self.address)
+  if not data[0].anyLoaded:
+    self.fetchOwnedCollectibles()
```

Comment:
here you're fetching the next page og collectibles for all addresses if the first address has not had its collectibles fetched yet. I'd fetch only the new addresses, something like
```
for i, addressData in data.pairs:
  if not addressData.anyLoaded:
    self.controller.fetchOwnedCollectibles(self.chainId, self.addresses[i])
```